### ISSUE TITLE
How to find the best model？

### ISSUE THREAD

**Issue by dj1367250510** [goldoak/DMSR#6]
Hello, I use the pre -training model you provided to achieve the effect of approaching your original text. But I trained your code and finally got a total of 120 models. When I used the 120th model to verify, I found that the results were poor, so I would like to ask if there is any way to find the best model to verify? Will the result of the training model converged? I need to verify each model by myself to get the best results?

**Comment by goldoak** [comment]
Hi there, thank you for trying our dataset and code. Here are answers to your questions:
1. We follow the same way as [SGPA](https://github.com/goldoak/DMSR/blob/master/train.py) to find the best model.
2. Of course the result will converge after training. You can view the training process using tensorboard.
3. No, you don't have to verify each model by yourself. Please refer to the first answer.

**Comment by dj1367250510** [comment]
> Hi there, thank you for trying our dataset and code. Here are answers to your questions:
> 
> 1. We follow the same way as [SGPA](https://github.com/goldoak/DMSR/blob/master/train.py) to find the best model.
> 2. Of course the result will converge after training. You can view the training process using tensorboard.
> 3. No, you don't have to verify each model by yourself. Please refer to the first answer.

Thank you for your previous answer, based on the results of the training visualization, I chose the model with the lowest training loss to validate the results, but the results are not as good as I would like. I would like to get some help from you, could you please guide me?
I have some questions:
1. on both datasets I performed the training time's only took about half an hour, is this consistent with you?
2. Just take the REAL dataset as an example, I spent a long time on the PNP calculation during the validation, which caused me to spend about 40 minutes on the validation, but I spent less time on the validation with the model you provided and the results were better, have you ever encountered such a situation at that time?
![image](https://github.com/goldoak/DMSR/assets/74551357/c8ed3664-887c-4832-b6fd-3449c38305d9)
![image](https://github.com/goldoak/DMSR/assets/74551357/f75fd997-b29a-4e00-a0ed-a28ebcd4c522)

**Comment by dj1367250510** [comment]
@goldoak 
Hello, I found the following two issues while debugging your training code:
1.If you are training CAMER,the length of traing_dataloader should be around 249127, but in the following code: for i, data in tqdm(enumerate(train_dataloader, 1), total=len(train_dataloader)):, I found that at this point train_dataloader is 4.
2. When printing the training loss, the following code: if i % 10 == 0, here the loss can not be printed, because i%10==0 is an error dead loop.

Translated with DeepL.com (free version)
![image](https://github.com/goldoak/DMSR/assets/74551357/961cf1e0-28f5-4fb0-a939-9d4baae1880f)
![image](https://github.com/goldoak/DMSR/assets/74551357/c8518547-723e-4ac4-8e20-6833d0e5b401)

**Comment by goldoak** [comment]
Thank you for your feedback! To solve those two issues, you only need to reset `train_steps` to a larger value. I set it to a smaller value for debugging.
https://github.com/goldoak/DMSR/blob/45353a80611e4b7d97208c381f59929791036213/train.py#L66

**Comment by dj1367250510** [comment]
> Thank you for your feedback! To solve those two issues, you only need to reset `train_steps` to a larger value. I set it to a smaller value for debugging.
> 
> https://github.com/goldoak/DMSR/blob/45353a80611e4b7d97208c381f59929791036213/train.py#L66

Thanks for your reply, do I need to change maxepoch if I reset train_steps = 4000?

**Comment by goldoak** [comment]
You can first try the default value written in the code. If it is not good then you have to adjust it accordingly, after all, it is a hyper-parameter.

**Comment by dj1367250510** [comment]
> 您可以先尝试代码中编写的默认值。如果它不好，那么你必须相应地调整它，毕竟它是一个超参数。

Thanks for your help, do you have the final version of the train.py file there?

**Comment by goldoak** [comment]
I just updated the `train.py`. Please try it first before you ask new questions. I will close this issue now.
